packages/openapi3/src/lib.ts

```diff
@@ -53,10 +53,55 @@ const EmitterOptionsSchema: JSONSchemaType<OpenAPI3EmitterOptions> = {
   type: "object",
   additionalProperties: false,
   properties: {
-    "file-type": { type: "string", enum: ["yaml", "json"], nullable: true },
-    "output-file": { type: "string", nullable: true },
-    "new-line": { type: "string", enum: ["crlf", "lf"], default: "lf", nullable: true },
-    "omit-unreachable-types": { type: "boolean", nullable: true },
+    "file-type": {
+      type: "string",
+      enum: ["yaml", "json"],
+      nullable: true,
+      description:
+        "If the content should be serialized as YAML or JSON. Default 'yaml', it not specified infer from the `output-file` extension",
+    },
+    "output-file": {
+      type: "string",
+      nullable: true,
+      description: [
+        "Name of the output file.",
+        " Output file will interpolate the following values:",
+        "  - service-name: Name of the service if multiple",
+        "  - version: Version of the service if multiple",
+        "",
+        ' @default `{service-name}.{version}.openapi.yaml` or `.json` if `file-type` is `"json"`',
+        "",
+        " @example Single service no versioning",
+        "  - `openapi.yaml`",
+        "",
+        " @example Multiple services no versioning",
+        "  - `openapi.Org1.Service1.yaml`",
+        "  - `openapi.Org1.Service2.yaml`",
+        "",
+        " @example Single service with versioning",
+        "  - `openapi.v1.yaml`",
+        "  - `openapi.v2.yaml`",
+        "",
+        " @example Multiple service with versioning",
+        "  - `openapi.Org1.Service1.v1.yaml`",
+        "  - `openapi.Org1.Service1.v2.yaml`",
+        "  - `openapi.Org1.Service2.v1.0.yaml`",
+        "  - `openapi.Org1.Service2.v1.1.yaml`    ",
```

Comment:
Oh these are just in the JSON directly as such. So I guess we can format however we want. Or do you imagine something consuming these "tags"?

Reply:
Hhm yeah they are consumed exactly as such. Just blindly copied the jsdoc comment. 

My ideal for this is that we can use `typespec` to define the emitter options. In the meantime i'll update to the markdown I want to see